### PR TITLE
Add capi cluster backup and restore role

### PR DESCRIPTION
**Pull request by sd109**

### CODE REVIEW

**Review by mkjpryor — CHANGES_REQUESTED** [review]

File: roles/velero/tasks/install.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  ansible.builtin.command: git clone "{{ velero_k8s_snapshot_controller_repo }}" external-snapshotter
+  when: not external_snapshotter_checkout.stat.exists
+
+- name: Ensure CSI snapshot controller is deployed
+  # Adapted from https://kubernetes-csi.github.io/docs/snapshot-controller.html
+  ansible.builtin.shell: |
+    cd ./external-snapshotter
+    git checkout "{{ velero_k8s_snapshot_controller_git_version }}"
+    kubectl kustomize client/config/crd | kubectl apply -f -
+    kubectl -n kube-system kustomize deploy/kubernetes/snapshot-controller | kubectl apply -f -
```
[the remark this answers is not in the document]
Comment (sd109) [reply roles/velero/tasks/install.yml:17]: It doesn't look like it. The controllers [readme](https://github.com/kubernetes-csi/external-snapshotter#usage) states

> it is strongly recommended that Kubernetes distributors bundle and deploy the controller and CRDs as part of their Kubernetes cluster management process

but doesn't say how it expects us to do that. I guess we could package it in capi-helm-charts but that's probably just side-stepping the problem.

File: roles/velero/tasks/restore.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    NAMESPACE=azimuth
+    kubectl -n $NAMESPACE exec consul-server-0 -- consul leave
+    sleep 2
+    kubectl -n $NAMESPACE rollout status statefulset/consul-server --watch
+    kubectl -n $NAMESPACE exec consul-server-1 -- consul leave
+    sleep 2
+    kubectl -n $NAMESPACE rollout status statefulset/consul-server --watch
+    kubectl -n $NAMESPACE exec consul-server-2 -- consul leave
+    sleep 2
+    kubectl -n $NAMESPACE rollout status statefulset/consul-server --watch
```
[the remark this answers is not in the document]
Comment (sd109) [reply roles/velero/tasks/restore.yml:38]: Good catch, but yes sadly `rollout restart` isn't enough. I'll adapt the script to work with a variable number of consul servers.

File: roles/velero/tasks/install.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkjpryor) [reply roles/velero/tasks/install.yml:17]: That actually feels a lot more like what we want to do...

File: roles/velero/tasks/restore.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkjpryor) [reply roles/velero/tasks/restore.yml:38]: The sooner we drop Consul the better...

File: roles/velero/tasks/install.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sd109) [reply roles/velero/tasks/install.yml:17]: Long term yes, I agree. Probably not a quick bit of work though. Would we just convert the yaml resources that are being applied here into `Manifest` addons? Or would you want to do it some other way? (e.g. package it as a Helm chart)

[the remark this answers is not in the document]
Comment (sd109) [reply roles/velero/tasks/install.yml:17]: Actually there is this https://gitlab.cern.ch/kubernetes/automation/charts/snapshot-controller/-/tree/master/snapshot-controller

[the remark this answers is not in the document]
Comment (mkjpryor) [reply roles/velero/tasks/install.yml:17]: Also this, which actually ships a chart into a repo we can use

https://github.com/piraeusdatastore/helm-charts/tree/main/charts/snapshot-controller

Actually, that datastore also looks kind of interesting...

[the remark this answers is not in the document]
Comment (sd109) [reply roles/velero/tasks/install.yml]: Switched to this piraeus chart for now

**Pull request merged**